structure_tests/structure_test.go

```diff
@@ -110,48 +29,64 @@ func compileAndRunRegex(regex string, base string, t *testing.T, err string, sho
 	}
 }
 
-func validateCommandTest(t *testing.T, tt CommandTest) {
-	if tt.Name == "" {
-		t.Fatalf("Please provide a valid name for every test!")
-	}
-	if tt.Command == "" {
-		t.Fatalf("Please provide a valid command to run for test %s", tt.Name)
+func Parse(fp string) (StructureTest, error) {
+	testContents, err := ioutil.ReadFile(fp)
+	if err != nil {
+		return nil, err
 	}
-	t.Logf("COMMAND TEST: %s", tt.Name)
-}
 
-func validateFileExistenceTest(t *testing.T, tt FileExistenceTest) {
-	if tt.Name == "" {
-		t.Fatalf("Please provide a valid name for every test!")
-	}
-	if tt.Path == "" {
-		t.Fatalf("Please provide a valid file path for test %s", tt.Name)
+	var unmarshal Unmarshaller
+	var versionHolder SchemaVersion
+
+	switch {
+	case strings.HasSuffix(fp, ".json"):
+		unmarshal = json.Unmarshal
+	case strings.HasSuffix(fp, ".yaml"):
+		unmarshal = yaml.Unmarshal
+	default:
+		return nil, errors.New("Please provide valid JSON or YAML config file.")
 	}
-	t.Logf("FILE EXISTENCE TEST: %s", tt.Name)
-}
 
-func validateFileContentTest(t *testing.T, tt FileContentTest) {
-	if tt.Name == "" {
-		t.Fatalf("Please provide a valid name for every test!")
+	if err := unmarshal(testContents, &versionHolder); err != nil {
+		return nil, err
 	}
-	if tt.Path == "" {
-		t.Fatalf("Please provide a valid file path for test %s", tt.Name)
+
+	version := versionHolder.SchemaVersion
+	if version == "" {
+		return nil, errors.New("Please provide JSON schema version.")
+	} else {
+		st := schemaVersions[version]
+		if st == nil {
+			return nil, errors.New("Unsupported schema version: " + version)
+		}
+		unmarshal(testContents, st)
+		tests, ok := st.(StructureTest) //type assertion
+		if !ok {
+			return nil, errors.New("Error encountered when type casting Structure Test interface!")
+		}
+		return tests, nil
 	}
-	t.Logf("FILE CONTENT TEST: %s", tt.Name)
 }
 
 var configFiles arrayFlags
 var tests StructureTest
 
-func init() {
+func TestMain(m *testing.M) {
 	flag.Var(&configFiles, "config", "path to the .yaml file containing test definitions.")
 	flag.Parse()
 
 	if len(configFiles) == 0 {
 		configFiles = append(configFiles, "/workspace/structure_test.json")
 	}
 
-	if err := Parse(configFiles, &tests); err != nil {
-		log.Fatalf("Error parsing config file: %s", err)
+	var err error
+	for _, file := range configFiles {
+		if tests, err = Parse(file); err != nil {
+			log.Fatalf("Error parsing config file: %s", err)
+		}
+		log.Printf("Running tests for file %s", file)
+		if exit := m.Run(); exit != 0 {
```

Comment:
Sorry for the back and forth, but it looks like the way you have this setup, we'll exit after the first failure. I think we're technically only supposed to call m.Run once as well.

Could you move some of this looping up into TestAll? That way we can run all the tests and only fail at the end.